### DISCUSSION TITLE
I want to use a CSS variable for the second argument (declaration-value) of the var function.

### DISCUSSION THREAD

**Discussion by kght6123** [tailwindlabs/tailwindcss#7783] [Help] [answered]
Thanks to the very nice Tailwind CSS framework.(≧∇≦)b

If you want to dynamically change the color of CSS with CSS variables, you may set the default value in the second argument when the first argument of the var function is invalid.
(Like [2] in the sample)

![スクリーンショット 2022-03-08 10 00 57](https://user-images.githubusercontent.com/24556516/157144852-aade02de-f8f4-4807-aee9-a0c954838889.png)
- [Taiwind PLAY Share Link](https://play.tailwindcss.com/hYOEJzY4pk)

If you try to achieve this with Tailwind CSS, you need to write it in color code as in [4].

If you specify it with a color code, it will not be reflected even if you change the theme, so I want to use a CSS variable.

Like [3] [5] in the sample, it doesn't work with pure CSS, but it would be helpful if you can use the default variable for the second argument in Tailwind CSS.

Is there any other good way?
Is there no choice but to define the default CSS variable in JavaScript framework etc.?

**Comment by skimhugo** [comment, marked as answer]
I think color example is not a good example in your scenario. Because tailwind's native theme solution is good enough. I do not see benefit in your solution, but see a little some fallback actually.

But css variable using is definitely a good topic. Because its native reactive feature.

**Comment by kght6123** [comment]
@skimmilk6877 
thank you. (≧∇≦)b
I will realize it by dynamically switching the class name (`bg-gray-500` and` bg-red-500` etc.) with JavaScript and setting the theme (class name) of Tailwind CSS.
https://tailwindcss.com/docs/theme

```html
  <button 
    class="bg-gray-500 p-4 rounded block"
    >bg-gray-500</button
  >
```

```html
  <button 
    class="bg-red-500 p-4 rounded block"
    >bg-red-500</button
  >
```

I was replacing the existing CSS Module code with Tailwind CSS, but I was overwhelmed by the dynamic color code handling.

**Reply by skimhugo** [reply]
Recently, I'm studing https://github.com/saadeghi/daisyui. I think what you want can be done by tailwind plugin system.

And I think daisyui make a really good job.
